File: stocks-frontend/src/app/models/gerente.ts
export interface Gerente {
  id?: number;
  nombre: string;
  apellido: string;
  email: string;
  telefono: string;
  identificacion: string;
  estado: 'ACTIVO' | 'INACTIVO' | 'SUSPENDIDO';
  rol: 'admin' | 'gerente' | 'supervisor';
  fechaCreacion?: string;
  fechaUltimaActualizacion?: string;
  idPlaza?: number;
  nombrePlaza?: string;
}

export interface CreateGerenteRequest {
  nombre: string;
  apellido: string;
  email: string;
  password: string;
  telefono: string;
  identificacion: string;
  rol: 'admin' | 'gerente' | 'supervisor';
  idPlaza?: number;
}

export interface UpdateGerenteRequest {
  nombre?: string;
  apellido?: string;
  telefono?: string;
  estado?: string;
  rol?: 'admin' | 'gerente' | 'supervisor';
  idPlaza?: number;
}

File: stocks-backend/src/main/java/com/kodehaus/stocksbackend/model/Gerente.java
package com.kodehaus.stocksbackend.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "gerente")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Gerente {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String nombre;

    @Column(nullable = false)
    private String apellido;

    @Column(nullable = false, unique = true)
    private String email;

    @Column(nullable = false)
    private String password;

    @Column(nullable = false)
    private String telefono;

    @Column(nullable = false)
    private String identificacion;

    @Column(nullable = false)
    private String estado; // ACTIVO, INACTIVO, SUSPENDIDO

    @Column(nullable = false)
    private String rol; // admin, gerente, supervisor

    @Column(name = "fecha_creacion", nullable = false)
    private LocalDateTime fechaCreacion;

    @Column(name = "fecha_ultima_actualizacion")
    private LocalDateTime fechaUltimaActualizacion;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_plaza", unique = true)
    private Plaza plaza;

    @PrePersist
    protected void onCreate() {
        fechaCreacion = LocalDateTime.now();
        if (estado == null) {
            estado = "ACTIVO";
        }
        if (rol == null) {
            rol = "gerente";
        }
    }

    @PreUpdate
    protected void onUpdate() {
        fechaUltimaActualizacion = LocalDateTime.now();
    }
}
